packages/suite/src/components/suite/DeviceDisplay.tsx

```diff
@@ -0,0 +1,209 @@
+                    <Text isPixelType={isPixelType} key={chunkIndex}>
+                        {chunk}
+                    </Text>
+                ))}
+            </Row>
+        ));
+
+    const renderChunks = (address: string) => {
+        const chunks = address.match(/.{1,4}/g);
+
+        // Add pagination icons
+        // @ts-expect-error
+        chunks?.splice(15, 0, <Icon size={iconSize} icon={iconNextName} color={iconColor} />);
+        // @ts-expect-error
+        chunks?.splice(16, 0, <Icon size={iconSize} icon={iconContinuesName} color={iconColor} />);
+
+        const groupedChunks = chunks?.reduce((result, chunk, index) => {
+            const rowIndex = Math.floor(index / 4);
+
+            if (!result[rowIndex]) {
+                result[rowIndex] = [];
+            }
+
+            result[rowIndex].push(chunk);
+
+            return result;
+        }, [] as ReactNode[][]);
+
+        if (isPaginated) {
+            const PAGE_SIZE = 4;
+            return (
+                <Content>
+                    <Chunks isPixelType={isPixelType}>
+                        {showChunksInRows(groupedChunks?.slice(0, PAGE_SIZE))}
+                    </Chunks>
+                    <Wrapper>
+                        <Divider isPixelType={isPixelType} />
+                        <AddressLabel isPixelType={isPixelType}>
+                            <Translation id="NEXT_PAGE" />
+                        </AddressLabel>
+                    </Wrapper>
+                    <Chunks isPixelType={isPixelType}>
+                        {showChunksInRows(groupedChunks?.slice(PAGE_SIZE), true)}
+                    </Chunks>
+                </Content>
+            );
+        }
+
+        return <Chunks isPixelType={isPixelType}>{showChunksInRows(groupedChunks)}</Chunks>;
+    };
+
+    const renderOriginal = (address: string) => {
+        if (isPaginated) {
+            const breakpoint = isPixelType ? 70 : 81;
+            return (
+                <Content>
+                    <Text isPixelType={isPixelType}>{address.slice(0, breakpoint)}</Text>
+                    <StyledNextIcon
+                        isPixelType={isPixelType}
+                        size={iconSize}
+                        icon={iconNextName}
+                        color={iconColor}
+                    />
+                    <Wrapper>
+                        <Divider isPixelType={isPixelType} />
+                        <AddressLabel isPixelType={isPixelType}>
+                            <Translation id="NEXT_PAGE" />
+                        </AddressLabel>
+                    </Wrapper>
+                    <StyledContinuesIcon
+                        isPixelType={isPixelType}
+                        size={iconSize}
+                        icon={iconContinuesName}
+                        color={iconColor}
+                    />
+                    <Text isPixelType={isPixelType}>
+                        &nbsp;&nbsp;&nbsp;{address.slice(breakpoint)}
```

Comment:
Just found out [this](https://developer.mozilla.org/en-US/docs/Web/CSS/text-indent) exists! :D  Please use it.